docs/networking.md

```diff
@@ -0,0 +1,175 @@
+    link/ether 72:54:12:ff:ce:6f brd ff:ff:ff:ff:ff:ff
+root@hyper1:~# ip link show ich1
+79: ich1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+    link/ether 06:7a:fc:1b:9a:67 brd ff:ff:ff:ff:ff:ff
+```
+
+Set the tap devices to up state:
+
+```bash
+root@hyper1:~# ip link set up ich0
+root@hyper1:~# ip link set up ich1
+
+root@hyper1:~# ip link show ich0
+78: ich0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN mode DEFAULT group default qlen 1000
+    link/ether 72:54:12:ff:ce:6f brd ff:ff:ff:ff:ff:ff
+root@hyper1:~# ip link show ich1
+79: ich1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN mode DEFAULT group default qlen 1000
+    link/ether 06:7a:fc:1b:9a:67 brd ff:ff:ff:ff:ff:ff
+```
+
+## Connect tap devices
+
+Different networking models can be used to provide external connectivity. In this example we will
+use 2 linux bridges emulating 2 different networks. The integration bridge (ich-int) in this example will also be used
+for external connectivity.
+
+Create the bridges and connect the cloud-hypervisor tap devices to the bridges:
+
+```bash
+root@hyper1:~# brctl addbr ich-int
+root@hyper1:~# brctl addbr ich-dpl
+root@hyper1:~# ip link set up ich-int
+root@hyper1:~# ip link set up ich-dpl
+root@hyper1:~# brctl addif ich-int ich0
+root@hyper1:~# brctl addif ich-dpl ich1
+root@hyper1:~# brctl show
+bridge name     bridge id               STP enabled     interfaces
+ich-dpl         8000.067afc1b9a67       no              ich1
+ich-int         8000.725412ffce6f       no              ich0
+```
+This completes the layer 2 wiring: The cloud-hypervisor is now connected to the hypervisor host via the 2 linux bridges.
+
+## IP (Layer 3) provisioning
+
+### Hypervisor host
+
+On the hypervisor host add the network gateway IP address of each network to the 2 linux bridges:
+
+```bash
+root@hyper1:~# ip addr add 192.168.4.1/24 dev ich-int
+root@hyper1:~# ip addr add 10.0.1.1/24 dev ich-dpl
+```
+The routing table of the hypervisor host should now also have corresponding routing entries:
+
+```bash
+root@hyper1:~# route -n
+Kernel IP routing table
+Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
+0.0.0.0         192.168.178.1   0.0.0.0         UG    600    0        0 wlan1
+10.0.1.0        0.0.0.0         255.255.255.0   U     0      0        0 ich-dpl
+192.168.4.0     0.0.0.0         255.255.255.0   U     0      0        0 ich-int
+192.168.178.0   0.0.0.0         255.255.255.0   U     600    0        0 wlan1
+```
+### Virtual Machine
+
+Within the virtual machine set the vNIC's to up state and provision the corresponding IP addresses on the 2 vNIC's:
+
+```bash
+root@cloud-hypervisor:~# ip link set up enp0s2
+root@cloud-hypervisor:~# ip link set up enp0s3
+root@cloud-hypervisor:~# ip addr add 192.168.4.2/24 dev enp0s2
+root@cloud-hypervisor:~# ip addr add 192.168.4.2/24 dev enp0s3
```

Comment:
I think you wanted:

```
ip addr add 10.0.1.2/24 dev enp0s3
```

Reply:
@sboeuf Given that we're passing an IP to the `--net` parameter, should'nt we set the tap IP address ourselves?

Reply:
regarding the type: sure. will change this.
For testing purposes setting an ip on the tap by CH is fore sure the easiest. But in real deployment scenario's is this really the case? As far as I recall qemu doesn't set this either. What if CH is connected to an ovs bridge providing DHCP by SDN etc? Then the IP on the tap isn't relevant or do you see this differently?

Reply:
@ericooper I think you're right, let's keep it this way.